src/win/fs.c

```diff
@@ -1064,16 +1064,7 @@ void fs__write(uv_fs_t* req) {
 }
 
 
-void fs__rmdir(uv_fs_t* req) {
-  int result = _wrmdir(req->file.pathw);
-  if (result == -1)
-    SET_REQ_WIN32_ERROR(req, _doserrno);
-  else
-    SET_REQ_RESULT(req, 0);
-}
-
-
-void fs__unlink(uv_fs_t* req) {
+void __unlink_rmdir(uv_fs_t* req, BOOL isrmdir) {
```

Comment:
If you are able to delete directories with files in them (as you seem to have encountered here https://github.com/JuliaLang/julia/pull/53394#issuecomment-1955828648), then (a) that is odd and (b) I wonder what flag we set that caused that

Looking at how cygwin implements a semi-reliable rmdir (in only 1000 lines of various well-commented common edge cases!), it seems it should be expected to fail:
https://github.com/cygwin/cygwin/blob/c90b20192dda446542a49578b9a74d8ee47f032c/winsup/cygwin/syscalls.cc#L952-L959